### PR TITLE
Adopting Swift 3 in Omise Swift

### PR DESCRIPTION
**Pull request by pitiphong-p**

### CODE REVIEW

File: OmiseSwift/Destroyable.swift [hunk 1]
```diff
@@ -5,20 +5,20 @@ public protocol Destroyable { }
 public extension Destroyable where Self: ResourceObject {
     public typealias DestroyOperation = Operation<Self>
     
-    public static func destroyOperation(parent: ResourceObject?) -> DestroyOperation {
+    public static func destroyOperation(_ parent: ResourceObject?, id: String) -> DestroyOperation {
```
Comment (igroomgrim) [inline OmiseSwift/Destroyable.swift:8]: I should close my pr or not because it changed in this pr
https://github.com/chakrit/omise-swift/pull/23

Comment (pitiphong-p) [reply OmiseSwift/Destroyable.swift:8]: It's ok. Your PR should be merged first then I will rebase on top of your PR again.

File: OmiseSwift/Request.swift [hunk 2]
```diff
@@ -1,117 +1,114 @@
...
     
-    public let client: Client
-    public let operation: Op
-    public let callback: Op.Callback?
+    open let client: Client
+    open let operation: Op
+    open let callback: Op.Callback?
     
-    public init(client: Client, operation: Op, callback: Op.Callback?) {
+     public init(client: Client, operation: Op, callback: Op.Callback?) {
```
Comment (igroomgrim) [inline OmiseSwift/Request.swift]: Whitespace

File: OmiseSwiftTests/ChargesOperationFixtureTests.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
+//
+//  ChargesOperationFixtureTests.swift
+//  OmiseSwift
+//
+//  Created by Pitiphong Phongpattranont on 8/29/2559 BE.
+//  Copyright © 2559 Omise. All rights reserved.
+//
+
```
Comment (igroomgrim) [inline OmiseSwiftTests/ChargesOperationFixtureTests.swift]: File header krub

File: OmiseSwiftTests/DisputeOperationFixtureTests.swift [hunk 4]
```diff
@@ -0,0 +1,132 @@
+//
+//  DisputeOperationFixtureTests.swift
+//  OmiseSwift
+//
+//  Created by Pitiphong Phongpattranont on 8/29/2559 BE.
+//  Copyright © 2559 Omise. All rights reserved.
+//
+
```
Comment (igroomgrim) [inline OmiseSwiftTests/DisputeOperationFixtureTests.swift]: Here too

File: OmiseSwiftTests/FixtureClient.swift [hunk 5]
```diff
@@ -0,0 +1,107 @@
+//
+//  FixtureClient.swift
+//  OmiseSwift
+//
+//  Created by Pitiphong Phongpattranont on 8/29/2559 BE.
+//  Copyright © 2559 Omise. All rights reserved.
+//
+
```
Comment (igroomgrim) [inline OmiseSwiftTests/FixtureClient.swift]: Here too

File: OmiseSwiftTests/DisputeOperationFixtureTests.swift (hunk 4 above)
Comment (pitiphong-p) [reply OmiseSwiftTests/DisputeOperationFixtureTests.swift]: Uggggrrrrr all the time Orz

File: OmiseSwiftTests/FixtureTestCase.swift [hunk 6]
```diff
@@ -0,0 +1,23 @@
+//
+//  FixtureTestCase.swift
+//  OmiseSwift
+//
+//  Created by Pitiphong Phongpattranont on 8/29/2559 BE.
+//  Copyright © 2559 Omise. All rights reserved.
+//
+
```
Comment (igroomgrim) [inline OmiseSwiftTests/FixtureTestCase.swift]: Here too

File: OmiseSwiftTests/RefundOperationFixtureTests.swift [hunk 7]
```diff
@@ -0,0 +1,74 @@
+//
+//  RefundOperationFixtureTests.swift
+//  OmiseSwift
+//
+//  Created by Pitiphong Phongpattranont on 8/29/2559 BE.
+//  Copyright © 2559 Omise. All rights reserved.
+//
+
```
Comment (igroomgrim) [inline OmiseSwiftTests/RefundOperationFixtureTests.swift]: Here too

File: OmiseSwiftTests/TransferOperationFixtureTests.swift [hunk 8]
```diff
@@ -0,0 +1,110 @@
+//
+//  TransferOperationFixtureTests.swift
+//  OmiseSwift
+//
+//  Created by Pitiphong Phongpattranont on 8/29/2559 BE.
+//  Copyright © 2559 Omise. All rights reserved.
+//
+
```
Comment (igroomgrim) [inline OmiseSwiftTests/TransferOperationFixtureTests.swift]: Here too kub

File: OmiseSwift/Destroyable.swift (hunk 1 above)
Comment (igroomgrim) [reply OmiseSwift/Destroyable.swift:8]: 🆗  W8 you and @chakrit review my pr 😄

File: OmiseSwiftTests/DisputeOperationFixtureTests.swift (hunk 4 above)
Comment (igroomgrim) [reply OmiseSwiftTests/DisputeOperationFixtureTests.swift]: Haha just a little bit

File: OmiseSwift/Destroyable.swift (hunk 1 above)
Comment (pitiphong-p) [reply OmiseSwift/Destroyable.swift:8]: 👍

**Comment by pitiphong-p** [comment]
Updated from comments

**Review by chakrit — APPROVED** [review]

**Review by igroomgrim — APPROVED** [review]

**Pull request merged**